File: desktop/src/main/java/com/corrodinggames/rts/gameFramework/graphics/Texture.java
package com.corrodinggames.rts.gameFramework.graphics;

import android.graphics.Bitmap;
import com.corrodinggames.rts.game.ColorMode;
import com.corrodinggames.rts.gameFramework.GameEngine;

/* JADX INFO: renamed from: com.corrodinggames.rts.gameFramework.m.e */
/* JADX INFO: loaded from: game-lib.jar:com/corrodinggames/rts/gameFramework/m/e.class */
public class Texture {
    public Texture[] a;
    public Texture[] b;
    public Texture[] c;
    private static int x;
    public int d;
    public int e;
    public int f;
    public String g;
    public Integer h;
    public ShaderProgram i;
    public int[] j;
    public Bitmap k;
    public boolean m;
    public boolean n;
    public int p;
    public int q;
    public int r;
    public int s;
    public float t;
    public float u;
    public boolean v;
    public boolean l = true;
    public boolean o = false;
    boolean w = false;

    public Texture() {
        int i = x;
        x = i + 1;
        this.d = i;
        this.e = 1;
    }

    public Texture[] a(ColorMode colorMode) {
        if (colorMode == ColorMode.pureGreen) {
            return this.a;
        }
        if (colorMode == ColorMode.hueAdd) {
            return this.b;
        }
        if (colorMode == ColorMode.hueShift) {
            return this.c;
        }
        GameEngine.logColored("getTeamImageCache coloringMode:" + colorMode);
        return this.a;
    }

    public void a(ColorMode colorMode, Texture[] textureArr) {
        if (colorMode == ColorMode.pureGreen) {
            this.a = textureArr;
            return;
        }
        if (colorMode == ColorMode.hueAdd) {
            this.b = textureArr;
        } else if (colorMode == ColorMode.hueShift) {
            this.c = textureArr;
        } else {
            GameEngine.logColored("setTeamImageCache coloringMode:" + colorMode);
            this.a = textureArr;
        }
    }

    public void a(String str) {
        this.g = str;
    }

    public String a() {
        return this.g;
    }

    public Bitmap b() {
        return this.k;
    }

    public Texture c() {
        return this;
    }

    public void a(boolean z) {
        this.o = z;
        e();
    }

    public void b(boolean z) {
        this.w = z;
    }

    public boolean d() {
        return this.w;
    }

    protected void e() {
    }

    public boolean f() {
        return this.m;
    }

    public void a(Bitmap bitmap) {
        this.k = bitmap;
        this.p = this.k.b();
        this.q = this.k.c();
        g();
    }

    public void g() {
        this.r = this.p / 2;
        this.s = this.q / 2;
        this.t = this.p / 2.0f;
        this.u = this.q / 2.0f;
    }

    public void a(Texture texture) {
        texture.o = this.o;
        texture.p = this.p;
        texture.q = this.q;
        texture.r = this.r;
        texture.s = this.s;
        texture.t = this.t;
        texture.u = this.u;
    }

    @Override //
    /* JADX INFO: renamed from: h, reason: merged with bridge method [inline-methods] */
    public Texture clone() {
        Texture texture = new Texture();
        texture.o = this.o;
        if (this.k != null) {
            Bitmap bitmapA = this.k.a(this.k.d(), true);
            if (bitmapA == null) {
                throw new OutOfMemoryError("Failed to copy bitmap: " + this.k.d());
            }
            texture.a(bitmapA);
        }
        return texture;
    }

    public Texture a(int i, int i2, boolean z) {
        Texture texture = new Texture();
        texture.o = this.o;
        if (this.k != null) {
            Bitmap bitmapA = Bitmap.a(i, i2, this.k.d());
            texture.a(bitmapA);
            if (z) {
                for (int i3 = 0; i3 < bitmapA.b(); i3++) {
                    for (int i4 = 0; i4 < bitmapA.c(); i4++) {
                        bitmapA.a(i3, i4, this.k.a(i3, i4));
                    }
                }
            }
        }
        return texture;
    }

    public void i() {
        if (this.j == null) {
            j();
        }
    }

    public void j() {
        if (this.k == null && GameEngine.isNonAndroidVersion && !GameEngine.isJavaDesktopVersion) {
            return;
        }
        if (this.j == null) {
            this.j = new int[this.p * this.q];
        }
        this.k.a(this.j, 0, this.p, 0, 0, this.p, this.q);
    }

    public boolean k() {
        return true;
    }

    public int a(int i, int i2) {
        if (this.j != null) {
            return this.j[i + (i2 * this.p)];
        }
        return this.k.a(i, i2);
    }

    public void a(int i, int i2, int i3) {
        if (this.j != null) {
            this.j[i + (i2 * this.p)] = i3;
        } else {
            this.k.a(i, i2, i3);
        }
    }

    public int l() {
        return this.q;
    }

    public int m() {
        return this.p;
    }

    public void n() {
    }

    public void o() {
        if (this.k != null) {
            this.k = null;
        }
        if (this.w) {
            GameEngine.logColored("remove with keepInGPUMemory=true");
        }
    }

    public void p() {
        if (this.k == null && GameEngine.isNonAndroidVersion && !GameEngine.isPCOrIOSVersion) {
            return;
        }
        if (this.j != null) {
            this.k.b(this.j, 0, this.p, 0, 0, this.p, this.q);
            this.j = null;
        }
        this.e++;
    }

    public void q() {
    }

    public void r() {
        this.j = null;
    }

    public void s() {
        r();
    }

    public void t() {
    }

    public int u() {
        return this.p * this.q * 8;
    }

    public void v() {
        this.a = null;
        this.b = null;
        this.c = null;
        this.e++;
    }

    public void w() {
    }

    public void x() {
    }

    public void y() {
    }

    public void z() {
    }

    public boolean A() {
        return false;
    }

    public ShaderProgram B() {
        return this.i;
    }

    public void a(ShaderProgram shaderProgram) {
        this.i = shaderProgram;
    }
}

File: core/src/main/kotlin/com/corrodinggames/rts/gameFramework/graphics/Texture.kt
package com.corrodinggames.rts.gameFramework.graphics

import com.corrodinggames.rts.game.ColorMode
import com.corrodinggames.rts.gameFramework.GameEngine

open class Texture : Cloneable {
    @JvmField
    var a: Array<Texture>? = null

    @JvmField
    var b: Array<Texture>? = null

    @JvmField
    var c: Array<Texture>? = null

    @JvmField
    var d: Int = nextTextureId()

    @JvmField
    var e: Int = 1

    @JvmField
    var f: Int = 0

    @JvmField
    var g: String? = null

    @JvmField
    var h: Int? = null

    @JvmField
    var j: IntArray? = null

    private var committedArgbPixels: IntArray? = null
    private var argbPixelLoader: (() -> IntArray?)? = null
    private var orderedAlphaRequired: Boolean = false
    private var premultipliedAlpha: Boolean = false
    private var pixelRevision: Int = 0
    private var shaderProgram: ShaderProgram? = null

    @JvmField
    var m: Boolean = false

    @JvmField
    var n: Boolean = false

    @JvmField
    var p: Int = 0

    @JvmField
    var q: Int = 0

    @JvmField
    var r: Int = 0

    @JvmField
    var s: Int = 0

    @JvmField
    var t: Float = 0f

    @JvmField
    var u: Float = 0f

    @JvmField
    var v: Boolean = false

    @JvmField
    var l: Boolean = true

    @JvmField
    var o: Boolean = false

    @JvmField
    var w: Boolean = false

    open fun a(colorMode: ColorMode): Array<Texture>? =
        when (colorMode) {
            ColorMode.pureGreen -> a
            ColorMode.hueAdd -> b
            ColorMode.hueShift -> c
            else -> {
                GameEngine.logColored("getTeamImageCache coloringMode:$colorMode")
                a
            }
        }

    open fun a(colorMode: ColorMode, textures: Array<Texture>?) {
        when (colorMode) {
            ColorMode.pureGreen -> a = textures
            ColorMode.hueAdd -> b = textures
            ColorMode.hueShift -> c = textures
            else -> {
                GameEngine.logColored("setTeamImageCache coloringMode:$colorMode")
                a = textures
            }
        }
    }

    open fun a(name: String?) {
        g = name
    }

    open fun a(): String? = g

    open fun b(): IntArray? {
        i()
        return j
    }

    val argbPixelsCopy: IntArray?
        get() = loadArgbPixelsIfNeeded()?.clone()

    open fun c(): Texture = this

    open fun a(value: Boolean) {
        o = value
        e()
    }

    open fun b(value: Boolean) {
        w = value
    }

    open fun d(): Boolean = w

    protected open fun e() = Unit

    open fun f(): Boolean = m

    open fun g() {
        r = p / 2
        s = q / 2
        t = p / 2.0f
        u = q / 2.0f
    }

    open fun a(texture: Texture) {
        texture.o = o
        texture.m = m
        texture.p = p
        texture.q = q
        texture.r = r
        texture.s = s
        texture.t = t
        texture.u = u
    }

    public open override fun clone(): Texture {
        val texture = Texture()
        texture.o = o
        texture.m = m
        texture.p = p
        texture.q = q
        texture.g()
        val pixels = argbPixelsCopy
        if (pixels != null) {
            texture.j = pixels
            texture.committedArgbPixels = pixels.clone()
        }
        return texture
    }

    open fun a(width: Int, height: Int, copyPixels: Boolean): Texture {
        val texture = Texture()
        texture.o = o
        texture.m = m
        texture.p = width
        texture.q = height
        texture.g()
        if (copyPixels) {
            texture.j = IntArray(width * height)
            val copyWidth = minOf(width, p)
            val copyHeight = minOf(height, q)
            for (x in 0 until copyWidth) {
                for (y in 0 until copyHeight) {
                    texture.j!![x + (y * width)] = a(x, y)
                }
            }
            texture.committedArgbPixels = texture.j!!.clone()
            texture.m = containsTransparentPixels(texture.j!!)
        }
        return texture
    }

    open fun i() {
        if (j == null) {
            j()
        }
    }

    open fun j() {
        val committedPixels = loadArgbPixelsIfNeeded()
        if (j == null && committedPixels != null) {
            j = committedPixels.clone()
            return
        }
        if (j == null) {
            if (p <= 0 || q <= 0) {
                return
            }
            j = IntArray(p * q)
        }
    }

    open fun k(): Boolean = true

    open fun a(x: Int, y: Int): Int {
        loadArgbPixelsIfNeeded()
        val editable = j
        if (editable != null) {
            return editable[x + (y * p)]
        }
        val committed = committedArgbPixels
        if (committed != null) {
            return committed[x + (y * p)]
        }
        return 0
    }

    open fun a(x: Int, y: Int, color: Int) {
        loadArgbPixelsIfNeeded()
        val editable = j
        when {
            editable != null -> {
                editable[x + (y * p)] = color
                pixelRevision++
            }

            committedArgbPixels != null -> {
                committedArgbPixels!![x + (y * p)] = color
                pixelRevision++
            }

            else -> {
                if (p <= 0 || q <= 0) {
                    p = maxOf(p, x + 1)
                    q = maxOf(q, y + 1)
                    g()
                }
                committedArgbPixels = IntArray(p * q)
                committedArgbPixels!![x + (y * p)] = color
                pixelRevision++
            }
        }
    }

    open fun l(): Int = q

    open fun m(): Int = p

    open fun n() = Unit

    open fun o() {
        if (w) {
            GameEngine.logColored("remove with keepInGPUMemory=true")
        }
        argbPixelLoader = null
    }

    open fun p() {
        val editable = j
        if (editable != null) {
            committedArgbPixels = editable.clone()
            argbPixelLoader = null
            m = containsTransparentPixels(editable)
            pixelRevision++
        }
        e++
    }

    open fun q() = Unit

    open fun r() {
        j = null
    }

    open fun s() {
        r()
    }

    open fun t() = Unit

    open fun u(): Int = p * q * 8

    open fun v() {
        a = null
        b = null
        c = null
        pixelRevision++
        e++
    }

    open fun w() = Unit

    open fun x() = Unit

    open fun y() = Unit

    open fun z() = Unit

    open fun A(): Boolean = false

    open fun B(): ShaderProgram? = shaderProgram

    open fun a(shaderProgram: ShaderProgram?) {
        this.shaderProgram = shaderProgram
    }

    fun shaderProgram(): ShaderProgram? = B()

    fun setShaderProgram(shaderProgram: ShaderProgram?) = a(shaderProgram)

    fun setSourceName(name: String?) = a(name)

    fun sourceName(): String? = a()

    fun editablePixels(): IntArray? = b()

    fun resolvedTexture(): Texture = c()

    fun updateCenter() = g()

    fun copyTextureSettingsTo(texture: Texture) = a(texture)

    fun width(): Int = m()

    fun height(): Int = l()

    fun commitPixels() = p()

    fun discardEditPixels() = s()

    fun releaseTexture() = o()

    fun estimatedMemoryBytes(): Int = u()

    fun ensureLoaded() = w()

    fun setCommittedArgbPixels(pixels: IntArray) {
        j = null
        committedArgbPixels = pixels.clone()
        argbPixelLoader = null
        m = containsTransparentPixels(pixels)
        pixelRevision++
    }

    fun setArgbPixelLoader(loader: (() -> IntArray?)?) {
        if (j == null && committedArgbPixels == null) {
            argbPixelLoader = loader
        }
    }

    protected fun invalidateArgbPixelSnapshot(loader: (() -> IntArray?)? = null) {
        j = null
        committedArgbPixels = null
        argbPixelLoader = loader
        pixelRevision++
    }

    fun setPremultipliedAlpha(value: Boolean) {
        premultipliedAlpha = value
    }

    fun usesPremultipliedAlpha(): Boolean = premultipliedAlpha

    fun getPixelRevision(): Int = pixelRevision

    fun requireOrderedAlpha() {
        orderedAlphaRequired = true
    }

    fun requiresOrderedAlpha(): Boolean = orderedAlphaRequired

    companion object {
        private var x: Int = 0

        private fun containsTransparentPixels(pixels: IntArray): Boolean =
            pixels.any { (it ushr 24) != 0xff }

        private fun nextTextureId(): Int {
            val id = x
            x = id + 1
            return id
        }
    }

    private fun loadArgbPixelsIfNeeded(): IntArray? {
        j?.let { return it }
        committedArgbPixels?.let { return it }
        val loaded = argbPixelLoader?.invoke()
        argbPixelLoader = null
        if (loaded != null) {
            committedArgbPixels = loaded
            m = containsTransparentPixels(loaded)
        }
        return loaded
    }
}
